platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/internal/service/scopes/Scope.java

```diff
@@ -17,75 +17,131 @@
+     * {@link BuildSession} services and descendant scopes, but not vice versa.
      */
     interface BuildSession extends UserHome, CrossBuildSession {}
 
     /**
-     * These services are recreated when in continuous build and shared across all nested builds.
-     * They are closed when the build invocation is completed.
-     *
-     * <p>{@link BuildSession} and parent scope services are visible to {@link BuildTree} scope services, but not vice versa.</p>
+     * The scope of a single <em>build execution</em> within a {@link BuildSession build session}.
+     * <p>
+     * <em>Build tree</em> is another name for the build definition ({@code BuildDefinition}),
+     * which corresponds to composite build.
+     * <p>
+     * The <em>build tree state</em> holds the state for the entire build definition for a single build execution within a session.
+     * The build tree state is managed by the {@code BuildTreeState} class.
+     * An instance is created at the start of a build execution and discarded at the end of that execution.
+     * <p>
+     * {@link BuildSession} and its parent services are visible to {@link BuildTree} services and descendant scopes, but not vice versa.
      */
     interface BuildTree extends BuildSession {}
 
     /**
-     * These services are created once per {@code org.gradle.api.initialization.Settings} the beginning of the build invocation
-     * These services are closed at the end of the build invocation.
-     *
-     * <p>{@link BuildTree} and parent scope services are visible to {@link Build} scope services, but not vice versa.</p>
+     * The scope of a single <em>build</em> within a {@link BuildTree build tree}.
+     * <p>
+     * The <em>build state</em> holds the state for a <em>build</em> within the <em>build definition</em> for a single <em>build execution</em>,
+     * and is contained by the <em>build tree state</em>.
+     * <p>
+     * The build state is managed by the {@code BuildState} class.
+     * An instance is created for each build in the build definition, once per build execution and is discarded at the end of that execution.
+     * <p>
+     * There is one-to-one correspondence between a <em>build</em> and its {@code Settings}.
+     * However, the build state also contains the state exposed to init scripts,
+     * evaluation of which happens prior to the evaluation of settings.
+     * <p>
+     * {@link BuildTree} and parent services are visible to {@link Build} services and descendant scopes, but not vice versa.
      */
     interface Build extends BuildTree {}
 
     /**
-     * These services are created once per {@code org.gradle.api.invocation.Gradle} at the beginning of the build invocation.
-     * These services are closed at the end of the build invocation.
-     *
-     * <p>{@link Build} and parent scope services are visible to {@link Gradle} scope services, but not vice versa.</p>
+     * The scope of the {@code org.gradle.api.invocation.Gradle} instance state.
```

Comment:
This is hard to understand without a bit of context. I think we should explain in the "Build" scope that that's the point where service state turns into things like Gradle, Settings, Project (because until then we we talking about state like daemon process, user home and so on).

Reply:
Maybe we could add an example scope -> state mapping (next to the visibility tree) at the beginning, something like:

- Global scope -> daemon process
- UserHome & CrossBuildSession scopes --> ???
- BuildSession scope -> user home + ???
- BuildTree scope -> ???
- Build scope -> Settings object + init scripts
- Gradle scope -> Gradle object
- Project scope -> Project object

Reply:
Added a section in the `Scope` javadoc that maps scopes to user-facing concepts